packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -513,13 +515,40 @@ async function setupOrgBrowser(
   );
 }
 
+function showTelemetryMessage(context: vscode.ExtensionContext) {
+  const messageAlreadyPrompted = context.globalState.get(
+    TELEMETRY_GLOBAL_VALUE
+  );
+  if (!messageAlreadyPrompted) {
+    // Show the message and set telemetry to true;
+    const showButtonText = nls.localize('telemetry_legal_dialog_button_text');
+    const showMessage = nls.localize(
+      'telemetry_legal_dialog_message',
+      TELEMETRY_OPT_OUT_LINK
+    );
+    vscode.window
+      .showInformationMessage(showMessage, showButtonText)
+      .then(selection => {
+        // Open disable telemetry link
+        if (selection && selection === showButtonText) {
+          vscode.commands.executeCommand(
+            'vscode.open',
+            vscode.Uri.parse(TELEMETRY_OPT_OUT_LINK)
+          );
+        }
+      });
+    context.globalState.update(TELEMETRY_GLOBAL_VALUE, true);
+  }
+}
```

Comment:
Calling the showInformationMessage api in the utils package is what caused vscode to be confused about what extension was showing the message, so I moved that logic to the core extension. Lmk if you think this function belongs somewhere else in the src.

Reply:
Looks good - Just a reminder that this won't show a "null" extension in production, only in dev.

Reply:
😞